src/olympia/access/acl.py

```diff
@@ -22,16 +22,13 @@ def action_allowed(request, app, action):
     'Admin:%' is true if the user has any of:
     ('Admin:*', 'Admin:%s'%whatever, '*:*',) as rules.
     """
-    allowed = any(match_rules(group.rules, app, action) for group in
-                  getattr(request, 'groups', ()))
-    return allowed
+    return action_allowed_user(request.user, app, action)
 
 
 def action_allowed_user(user, app, action):
     """Similar to action_allowed, but takes user instead of request."""
-    allowed = any(match_rules(group.rules, app, action) for group in
-                  user.groups.all())
-    return allowed
+    return (user.is_authenticated() and any(
+        match_rules(group.rules, app, action) for group in user.groups_list))
```

Comment:
Yeah that's a good idea. I added that late - because `AnonymousUser` does not have my cached property - and didn't try to make it prettier. I like your version, I'll use that.